src/raft_server.c

```diff
@@ -1239,95 +1239,42 @@ void raft_offer_log(raft_server_t* me_, raft_entry_t* entries,
             me->voting_cfg_change_log_idx = raft_get_current_idx(me_);
 
         int node_id = me->cb.log_get_node_id(me_, raft_get_udata(me_),
-                                             ety, idx);
+                                             ety, idx + i);
         raft_node_t* node = raft_get_node(me_, node_id);
         int is_self = node_id == raft_get_nodeid(me_);
```

Comment:
Consider adding a new replica "node 2". On node 2, when we offer the entry that says "add nonvoting node 2", this raft_get_nodeid call will return -1, while node_id will be 2, correct?